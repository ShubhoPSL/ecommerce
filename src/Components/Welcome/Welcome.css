.base {
  @apply flex justify-center items-center flex-col
bg-yellow-300  h-[83.25vh];
}

.welcome {
  @apply text-4xl capitalize 
    font-medium;
}

.ecommerce {
  @apply text-4xl uppercase font-semibold;
}
