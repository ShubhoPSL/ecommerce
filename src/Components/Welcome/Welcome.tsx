import React from 'react'

function Welcome() {
    return (
        <div className='base'>
            <p className='welcome'>Welcome To </p>
            <h1 className='ecommerce'>
                ShopKart
            </h1>
        </ div >
    )
}

export default Welcome