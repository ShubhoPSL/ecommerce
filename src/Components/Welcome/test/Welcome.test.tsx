import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Welcome from "../Welcome";
import { RecoilRoot } from "recoil";

describe("Welcome Page Component Testing", () => {
  const NewWelcome: React.FC = () => {
    return (
      <RecoilRoot>
        <Welcome />
      </RecoilRoot>
    );
  };
  afterEach(() => {
    cleanup();
  });
  test("should have Welcome Message", () => {
    render(<NewWelcome />);
    expect(screen.getByText(/welcome/i)).toBeInTheDocument();
  });
  test("should contains heading with E-Commerce Shopping Site", () => {
    render(<NewWelcome />);
    expect(
      screen.getByRole("heading", { name: "ShopKart" })
    ).toBeInTheDocument();
  });
  test("should match with snapshot of welcome module", () => {
    const tree = renderer.create(<NewWelcome />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
