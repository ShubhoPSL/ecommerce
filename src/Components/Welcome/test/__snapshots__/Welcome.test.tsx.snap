// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page Component Testing should match with snapshot of welcome module 1`] = `
<div
  className="base"
>
  <p
    className="welcome"
  >
    Welcome To 
  </p>
  <h1
    className="ecommerce"
  >
    ShopKart
  </h1>
</div>
`;
